### PR TITLE
BE-6728

### PR DESCRIPTION
**Pull request by mkorolyov**
+ possibility to exclude files from parsing
+ changes to support new default layout.

### CODE REVIEW

File: generator.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ func NewGenerator(structs []StructDef) generator {
 func (g generator) Generate(path, outputFile string, needCheck bool) error {
 	pkg := filepath.Base(path)
 	if pkg == "." {
-		pkg = "main"
+		pkg = "api"
```
Comment (l1va) [inline generator.go]: ??? why

File: inspector.go [hunk 2]
```diff
@@ -63,17 +70,27 @@ func (insp *inspector) Result() []StructDef {
 	return res
 }
 
-func getFilesForInspect(dir, outputFile string) ([]string, error) {
-	if strings.HasPrefix(dir, "/") {
+func getFilesForInspect(cfg inspectorConfig) ([]string, error) {
+	dir := cfg.dir
+	if strings.HasPrefix(cfg.dir, "/") {
```
Comment (l1va) [inline inspector.go]: change cfg.dir -> dir please

File: main.go [hunk 3]
```diff
@@ -13,6 +13,8 @@ import (
 func main() {
 	outputFilePtr := flag.String("outputFile", "validators.go", "output file name")
 	needValidatableCheckPtr := flag.Bool("needValidatableCheck", true, "check struct on Validatable before calling Validate()")
+	excludeRegexp := flag.String("excludeRegexp", `(client\.go|client_mock\.go)`,
+		"comma separated list of file names that generator should exclude, e.g. `client.go;client_mock.go`")
```
Comment (l1va) [inline main.go]: comma separated? examples has `;` . And maybe rename flag to excludeFiles or excludeList ? it is not regexp from client view.

File: inspector.go [hunk 4]
```diff
@@ -10,6 +10,7 @@ import (
 	"strings"
 
 	"github.com/gojuno/genval/types"
+	"regexp"
```
Comment (mkabischev) [inline inspector.go]: please fix imports order

File: generator.go (hunk 1 above)
Comment (mkabischev) [inline generator.go]: I think we should create and think about `genval` as some standalone tool for generating validations and don\`t use our internal packages and directories

File: main.go (hunk 3 above)
Comment (mkabischev) [inline main.go]: Comment isn\`t clear, can I use regexp or it supports only `;` separated values? Or I can use several regexpes?

Comment (mkorolyov) [inline main.go]: updated

File: inspector.go (hunk 4 above)
Comment (mkorolyov) [inline inspector.go]: updated

File: generator.go (hunk 1 above)
Comment (mkorolyov) [inline generator.go]: fixed

**Pull request merged**